### PR TITLE
[refactor] #4039: Enhance ClientCli Wait Mechanism for Transaction Completion in pytests and Remove Fixed Sleeps 

### PR DESCRIPTION
**Pull request by AlexStroke**
### Linked issue

Closes #{issue_number} 

### Checklist

- [ ] I've read `CONTRIBUTING.md`
- [ ] I've used the standard signed-off commit format (or will squash just before merging)
- [ ] All applicable CI checks pass (or I promised to make them pass later)
- [ ] (optional) I've written unit tests for the code changes
- [ ] I replied to all comments after code review, marking all implemented changes with thumbs up

### CODE REVIEW

File: client_cli/pytests/common/settings.py [hunk 1]
```diff
@@ -11,11 +11,9 @@
 
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
 
-ROOT_DIR = os.environ.get("CLIENT_CLI_DIR", BASE_DIR)
-
 PATH_CONFIG_CLIENT_CLI = os.environ["CLIENT_CLI_CONFIG"]
 CLIENT_CLI_PATH = os.environ["CLIENT_CLI_BINARY"]
-PEERS_CONFIGS_PATH = os.path.join(ROOT_DIR, "peers_configs")
+PEERS_CONFIGS_PATH = os.path.join(BASE_DIR, "peers_configs")
```
Comment (0x009922) [inline client_cli/pytests/common/settings.py:16]: What this is for? I found it was introduced by #4036, but there is no documentation for it (please add to README).

File: client_cli/pytests/src/client_cli/configuration.py [hunk 2]
```diff
@@ -63,20 +66,25 @@ def generate_by_peers(self, peers_configs_dir):
...
             config_copy = self._config.copy()
-            config_copy["TORII_API_URL"] = f"http://localhost:{port}"
-            file_name = f"config_to_peer_{port}.json"
+            parsed_url = urlparse(config_copy["torii_url"])
+            updated_url = parsed_url._replace(
+                netloc=f"{parsed_url.hostname}:{port}"
+            ).geturl()
+            config_copy["torii_url"] = updated_url
+
+            file_name = f"config_to_peer_{port}.toml"
```
Comment (0x009922) [inline client_cli/pytests/src/client_cli/configuration.py:75]: Why this all is needed?

You can use the same client configuration file each time and override `TORII_API_URL` for each execution separately.

File: client_cli/pytests/src/client_cli/client_cli.py [hunk 3]
```diff
@@ -384,6 +429,50 @@ def _execute_single(self, command):
...
+        idx = 0
+        try:
+            while idx < len(output):
+                obj, idx_next = decoder.raw_decode(output[idx:])
+                if (
+                    obj.get("Pipeline", {}).get("entity_kind") == "Transaction"
+                    and obj.get("Pipeline", {}).get("status") == "Committed"
+                    and obj.get("Pipeline", {}).get("hash") == transaction_hash
+                ):
+                    return True
```
Comment (0x009922) [inline client_cli/pytests/src/client_cli/client_cli.py:470]: I am not sure about the version of Python we are using. Can't we use [pattern matching](https://peps.python.org/pep-0636/) here?

```suggestion
                match obj:
                    case { "Pipeline": { "entity_kind": "Transaction", "status": "Committed", "hash": hash } } if hash == transaction_hash:
                        return True
```

File: client_cli/pytests/common/settings.py (hunk 1 above)
Comment (AlexStroke) [reply client_cli/pytests/common/settings.py]: removed it

File: client_cli/pytests/src/client_cli/configuration.py (hunk 2 above)
Comment (AlexStroke) [reply client_cli/pytests/src/client_cli/configuration.py]: Agree. Removed it.

File: client_cli/pytests/src/client_cli/client_cli.py [hunk 4]
```diff
@@ -53,6 +59,37 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
         self.reset()
 
+    def start_listening_to_events(self, peers_ports):
+        """
+        Initializes listening to events on all peers.
+        """
+        self.transaction_status = {}
+        self.threads = []
+        for port in peers_ports:
+            self.config.update_torii_url(port)
```
Comment (Erigara) [inline client_cli/pytests/src/client_cli/client_cli.py:69]: This part looks like every thread will see only latest call to `update_torii_url`.

**Comment by AlexStroke** [comment]
Closed this draft because we decided to make better solution not in the test framework. https://github.com/hyperledger/iroha/issues/4753

**Pull request closed without merging**